virtualizarr/tests/test_codecs.py

```diff
@@ -1,161 +1,141 @@
-            manifest_array.zarray.dtype
+        # Create a Zarr array in memory with the codecs
+        zarr_array = zarr.create(
+            shape=(1000, 1000),
+            chunks=(100, 100),
+            dtype="int32",
+            store=None,
+            zarr_format=zarr_format,
+            # compressor=compressor,
+            # filters=filters,
+            codecs=codecs,
         )
-        assert actual_codecs == expected_codecs
-
-    @requires_zarr_python_v3
-    def test_manifest_array_zarr_v3(self):
-        """Test that get_codecs works for ManifestArray with Zarr v2 metadata."""
-        from zarr.codecs import BytesCodec  # type: ignore[import-untyped]
 
-        manifest_array = self.create_manifest_array(zarr_format=3)
+        # Populate the Zarr array with data
+        zarr_array[:] = np.arange(1000 * 1000).reshape(1000, 1000)
+        return zarr_array
 
-        # Get codecs and verify
+    def test_manifest_array_zarr_v3_default(self, create_manifestarray):
+        """Test get_codecs with ManifestArray using default v3 codec."""
+        manifest_array = create_manifestarray(codecs=None)
         actual_codecs = get_codecs(manifest_array)
         expected_codecs = tuple([BytesCodec(endian="little")])
         assert actual_codecs == expected_codecs
 
-    def create_zarr_array(
-        self,
-        compressor=None,
-        filters=None,
-        codecs=None,
-        zarr_format=2,
-    ):
-        import zarr  # type: ignore[import-untyped]
-
-        shared_kwargs = {
-            "shape": (1000, 1000),
-            "chunks": (100, 100),
-            "dtype": "int32",
-            "store": None,
-            "zarr_format": zarr_format,
-        }
-        if zarr_format == 2:
-            shared_kwargs["compressor"] = compressor
-            shared_kwargs["filters"] = filters
-        elif zarr_format == 3:
-            shared_kwargs["codecs"] = codecs
-        # Create a Zarr array in memory with the codecs
-        zarr_array = zarr.create(**shared_kwargs)
-
-        # Populate the Zarr array with data
-        zarr_array[:] = np.arange(1000 * 1000).reshape(1000, 1000)
-
-        return zarr_array
-
-    @requires_zarr_python_v3
-    def test_zarr_v2(self):
-        # Define your codecs (compressor and filters)
-        compressor = Blosc(cname="zstd", clevel=5, shuffle=Blosc.SHUFFLE)
-        filters = [Delta(dtype="int32")]
-
-        zarr_array_v2 = self.create_zarr_array(compressor=compressor, filters=filters)
-        # Test codecs
-        actual_codecs = get_codecs(zarr_array_v2)
-        expected_codecs = Codec(
-            compressor=Blosc(
-                cname="zstd", clevel=5, shuffle=Blosc.SHUFFLE, blocksize=0
-            ),
-            filters=[
-                Delta(dtype="<i4"),
-            ],
+    def test_manifest_array_zarr_v3_with_codecs(self, create_manifestarray):
+        """Test get_codecs with ManifestArray using multiple v3 codecs."""
+        test_codecs = [
```

Comment:
This could also be a global variable `TEST_CODECS`, or even a fixture providing multiple sets of test codecs. See also @dcherian's `codecs` hypothesis strategy that we could use in future (https://github.com/zarr-developers/zarr-python/pull/2822).

Reply:
I've implemented the codecs as global vars, and implemented a multiple scenario test in test_codecs with various codec scenarios in https://github.com/zarr-developers/VirtualiZarr/pull/429/commits/b5a1dc69b29da135e1021c3c994b34fe48071f48 - do you think this is a good solution in lieu of using hypothesis strategy (I need to read up more on `hypothesis`)

Reply:
That's great. The codecs are now parametrizable, which is what we need to generalize those parameters later to a whole hypothesis strategy.